pkg/sdkstats/sdkstats.go

```diff
@@ -2,161 +2,121 @@ package sdkstats
 )
 
-//RepoMetadata structure for storing Repository details
+// RepoMetadata struct is for storing individual epository details.
 type RepoMetadata struct {
-	FullName     string
+	URL          string
+	Name         string
+	Owner        string
 	ProjectType  string
 	Stars        int
-	SDKVersion   int
+	Version      string
 	CreatedAt    string
 	PushedAt     string
 	TotalCommits int
 }
 
-//GetStats returns []RepoMetaData populated with Code results from Github Search Code API, for a given search string
-func GetStats(client *github.Client, q string) ([]RepoMetadata, error) {
-	var repoList []RepoMetadata
-	var projectType string
-	var sdkVersion int
-
-	//Search function return CodeResults for the q queery, and also Basic repository data.
-	searchOp, err := Search(client, q)
-
-	if err != nil {
-		return nil, err
-	}
-
-	if strings.Contains(q, "helm") {
-		projectType = "helm"
-	}
-	if strings.Contains(q, "ansible") {
-		projectType = "ansible"
-	}
-	if strings.Contains(q, "go.mod") {
-		projectType = "go.mod"
-	}
-	if strings.Contains(q, "Gopkg.toml") {
-		projectType = "Gopkg.toml"
-	}
-	for j := 0; j < len(searchOp); j++ {
-		for i := 0; i < len(searchOp[j].CodeResults); i++ {
-			owner := searchOp[j].CodeResults[i].GetRepository().GetOwner().GetLogin()
-			name := searchOp[j].CodeResults[i].GetRepository().GetName()
-			fullName := searchOp[j].CodeResults[i].GetRepository().GetFullName()
-
-			//getVersion returns SDK version by doing Fragment search
-			if projectType == "Gopkg.toml" {
-				sdkVersion = 0
-			} else {
-				sdkVersion = getversion(searchOp[j].CodeResults[i], projectType)
-			}
+// RepoMetadataQuery struct provides search queries, projecttype, and VersionParser declaration
+type RepoMetadataQuery struct {
+	ProjectType   string
+	Queries       []string
+	VersionParser VersionParser
+}
 
-			//GetRepoDetails returns Repository specififc details
-			repoMap := GetRepoDetails(client, owner, name)
+// VersionParser Interface implements ParseVersion(), to parse version number from a given Text Match result.
+type VersionParser interface {
+	ParseVersion(codeResults github.CodeResult) (string, error)
+}
 
-			TotalCommits, err := strconv.Atoi(repoMap["TotalCommits"])
-			Stars, err := strconv.Atoi(repoMap["Stars"])
-			if err != nil {
-				return nil, err
-			}
+// GetStats returns List of Repositories satisfyng the Search criteria,
+// populated with Stars, TotalCommits, Initial and Last Commit details.
+// Also included are basic details of repository such as URL, Owner, and name.
 
-			repoDetails := &RepoMetadata{
-				FullName:     fullName,
-				ProjectType:  projectType,
-				SDKVersion:   sdkVersion,
-				Stars:        Stars,
-				CreatedAt:    repoMap["CreatedAt"],
-				PushedAt:     repoMap["PushedAt"],
-				TotalCommits: TotalCommits,
+func GetStats(ctx context.Context, tc *http.Client, rq RepoMetadataQuery) ([]RepoMetadata, error) {
+	var repoList []RepoMetadata
+	var repoDetails *RepoMetadata
```

Comment:
`repoDetails` doesn't need to be declared at this scope since it is only used in the loop.

In the loop, you can use the following to declare and initialize in a single statement:
```
repoDetails, err := getRepoDetails(...)
```

Reply:
done